### ISSUE TITLE
Expanding a discrete scale by an additive constant doesn't work for single level factors

### ISSUE THREAD

**Issue by gdmcdonald** [tidyverse/ggplot2#3012]
Similar to #2281 but apparently not fixed even with the latest version.

expand_scale(mult = 0, add = 2) doesn't add space around facets with only one discrete factor level in the facet. Doesn't seem like expected behaviour.

```r

ggplot(data = mpg,
       aes(y = model,
           x = cyl,
           color = year))+
  geom_point()+
  facet_grid(manufacturer~.,drop = T, scales = "free_y", space = "free_y")+
  scale_y_discrete(expand = expand_scale(mult = 0, add = 2))+
  theme(strip.text.y = element_text(angle = 0))

```

![mpg_plot](https://user-images.githubusercontent.com/20785842/48808296-85b4ad80-ed74-11e8-8c36-f713695e6cff.png)

**Comment by dpseidel** [comment]
This behavior was actually fixed in the scales 1.0.0 release.  See below with the newest release of both scales and ggplot2. 


``` r
library(ggplot2)
ggplot(data = mpg,
       aes(y = model,
           x = cyl,
           color = year))+
  geom_point()+
  facet_grid(manufacturer~.,drop = T, scales = "free_y", space = "free_y")+
  scale_y_discrete(expand = expand_scale(mult = 0, add = 2))+
  theme(strip.text.y = element_text(angle = 0))
```

![](https://i.imgur.com/0Xn7DNu.png)

``` r

packageVersion("scales")
#> [1] '1.0.0'
packageVersion("ggplot2")
#> [1] '3.1.0'
```

Created on 2018-11-20 by the [reprex package](http://reprex.tidyverse.org) (v0.2.0).

**Comment by gdmcdonald** [comment]
My apologies, didn't realise I needed to update the scales package.
